include/mesh/boundary_info.h

```diff
@@ -80,6 +80,14 @@ class BoundaryInfo : public ParallelObject
    */
   BoundaryInfo & operator=(const BoundaryInfo & other_boundary_info);
 
+  /**
+   * This tests for data equality via element ids
+   */
+  bool operator== (const BoundaryInfo & other_boundary_info);
+
+  bool operator!= (const BoundaryInfo & other_boundary_info) {
```

Comment:
This question applies to all the other comparison operators: can we mark these all (or at least some) as `const`?